File: Scripts/generated_codes/java/13.java
/*

Explanation

In the first test case, the Chef is at 0,1, the head server is at 0,0 and the sous-chef is at 1,0. The distance between every pair is at most 1, so they can all communicate.


In the second test case, the Chef is at 0,1, the head server is at 0,0 and the sous-chef is at 1,0. The distance between every pair is at most 2, so they can all communicate.


In the third test case, the Chef is at 0,0, the head server is at 0,2 and the sous-chef is at 2,1. Although the Chef can communicate directly with the head server, and the head server can communicate directly with the sous-chef, the Chef is too far away from the sous-chef to communicate directly. They cannot communicate indirectly either because no other transceiver is close enough to both the Chef and the sous-chef.


 */

import java.util.*;

public class ChefAndHisStaff {

    static Scanner s = new Scanner(System.in);
    static int MAX = 5000;
    static int[] dx = {-1, -1, -1, 0, 0, 1, 1, 1};
    static int[] dy = {1, 0, -1, 1, -1, 1, 0, -1};

    public static void main(String[] args) {
        int T = s.nextInt();
        while (T-- > 0) {
            int r = s.nextInt();
            int fx = s.nextInt();
            int fy = s.nextInt();
            int sx = s.nextInt();
            int sy = s.nextInt();
            int tx = s.nextInt();
            int ty = s.nextInt();

            double d1 = Math.sqrt(Math.pow(Math.abs(fx - sx), 2) + Math.pow(Math.abs(fy - sy), 2));
            double d2 = Math.sqrt(Math.pow(Math.abs(sx - tx), 2) + Math.pow(Math.abs(sy - ty), 2));
            double d3 = Math.sqrt(Math.pow(Math.abs(tx - fx), 2) + Math.pow(Math.abs(ty - fy), 2));

            if (d1 <= r && d2 <= r || d1 <= r && d3 <= r || d2 <= r && d3 <= r) {
                System.out.println("yes");
            } else if (d1 > r && d2 > r) {
                if (d3 <= r) {
                    System.out.println("yes");
                } else {
                    System.out.println("no");
                }
            } else if (d1 > r && d3 > r) {
                if (d2 <= r) {
                    System.out.println("yes");
                } else {
                    System.out.println("no");
                }
            } else if (d2 > r && d3 > r) {
                if (d1 <= r) {
                    System.out.println("yes");
                } else {
                    System.out.println("no");
                }
            }
        }
    }
}

File: Generated_Codes_Check/Results/python/13.py
"""


Explanation

Example case 1: The first transceiver is a distance 1 from the second and the second is a distance 1 from the third, so all three are
close enough to communicate.

Example case 2: The Chef and the head server are close enough to communicate directly, and the sous-chef is close enough to the
head server to communicate through it.

Example case 3: No two are close enough to communicate directly, and there is no transceiver close enough to all three to serve as an
intermediate device.



Hint:
Try to find the distance between two points using Pythagoras Theorem.

Solution:
"""
import math
t=int(input())
for _ in range(t):
  r=int(input())
  x1,y1=map(int,input().split())
  x2,y2=map(int,input().split())
  x3,y3=map(int,input().split())
  d1=math.sqrt(((x1-x2)**2)+((y1-y2)**2))
  d2=math.sqrt(((x3-x2)**2)+((y3-y2)**2))
  d3=math.sqrt(((x1-x3)**2)+((y1-y3)**2))
  if d1<=r and d2<=r or d1<=r and d3<=r or d2<=r and d3<=r:
    print("yes")
  else:
    print("no")